src/components/shared/ButtonWithPanel/ArrowPanel.js

```diff
@@ -52,8 +51,6 @@ export class ArrowPanel extends React.PureComponent<Props, State> {
 
       setTimeout(this._onCloseAnimationFinish(openGeneration), 400);
 
-      this.props.onClose();
-
```

Comment:
The proverb "sleep on it" as never been so true. Yesterday you suggested to use `requestAnimationFrame` here. But waking up this morning, I thought of this other solution of using `componentDidUpdate` to trigger the side-effects. This works fine, and I think this solution is also closer to how React is designed.

Note that I moved both callback calls for symmetry.

Reply:
Thought: Ah, I wonder if this has been a misconception on my part. I should re-examine my uses of rAF. I think my mind was a bit blurry on which life cycle methods were deprecated, and which not. I guess `componentDidUpdate` could be renamed `doSideEffectAfterUpdate`